src/promptflow-tracing/promptflow/tracing/_trace.py

```diff
@@ -41,6 +41,71 @@ def _record_cancellation_exceptions_to_span(span: Span):
         raise
 
 
+def handle_span_exception(span, exception):
+    if isinstance(span, Span) and span.is_recording():
+        # Record the exception as an event
+        span.record_exception(exception)
+        # Records status as error
+        span.set_status(
+            Status(
+                status_code=StatusCode.ERROR,
+                description=f"{type(exception).__name__}: {str(exception)}",
+            )
+        )
+
+
+def handle_output(span, inputs, output, trace_type):
+    if isinstance(output, (Iterator, AsyncIterator)):
+        # Set should_end to False to delay span end until generator exhaustion, preventing premature span end.
+        setattr(span, "__should_end", False)
+        output = Tracer.pop(output)
+        if isinstance(output, Iterator):
+            return traced_generator(span, inputs, output, trace_type)
+        else:
+            return traced_async_generator(span, inputs, output, trace_type)
+    else:
+        enrich_span_with_trace_type(span, inputs, output, trace_type)
+        span.set_status(StatusCode.OK)
+        return Tracer.pop(output)
+
+
+@contextlib.contextmanager
+def start_as_current_span(
```

Comment:
Shall we separate this PR into two, one is the update of the structure of trace related code (but no business changes), another is for the streaming LLM span update?

Reply:
Thank you for the suggestion. Given our upcoming release, it's crucial to maintain a clear focus on our changes. While separating the PR could enhance clarity, merging them in this PR ensures we streamline our workflow and avoid potential delays.